src/errors/error-utils.ts

```diff
@@ -0,0 +1,37 @@
+import {MomentoErrorCode, SdkError} from './errors';
+
+export abstract class ErrorBody {
+  protected _innerException: SdkError;
+  public message(): string {
+    return this._innerException.wrappedErrorMessage();
+  }
+
+  public innerException(): object {
+    return this._innerException;
+  }
+
+  public errorCode(): MomentoErrorCode {
+    return this._innerException.errorCode();
+  }
+
+  public toString(): string {
+    return this.message();
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function applyMixins(derivedCtor: any, constructors: any[]) {
```

Comment:
It is not immediately clear what this function is doing, might be good to add a comment about exactly what its purpose is. Also, could this function be typed with `export function applyMixins(derivedCtor: ErrorBody, constructors: ErrorBody[])`? Just looking at where its used it looks like it could potentially be typed this way.

Reply:
Agree maybe also just some JS Doc comments on what its doing and why we need this just for another dev coming in and looking at this?

Reply:
Fair point. I'll add some explanation.

Reply:
> Also, could this function be typed with `export function applyMixins(derivedCtor: ErrorBody, constructors: ErrorBody[])`? Just looking at where its used it looks like it could potentially be typed this way.

The class passed into `derivedCtor` is not of type `ErrorBody` and the `constructors` argument could be passed anything. This is intended to be a general purpose function for using [mixing](https://www.typescriptlang.org/docs/handbook/mixins.html#alternative-pattern) that we happen to be using for a single specific purpose for now.